src/exchanges/crdtExchange.ts

```diff
@@ -465,6 +458,64 @@ class CrdtManager {
     }
   }
 
+  private persistMutations() {
```

Comment:
It seems to me like the next thing we need to change in this PR is to make this function async and make every caller `await` this so that we don't proceed further until we know that the write is complete.